agent/app/agent.go

```diff
@@ -515,7 +515,9 @@ func (agent *ecsAgent) registerContainerInstance(
 // from a check point.
 func (agent *ecsAgent) reregisterContainerInstance(client api.ECSClient,
 	capabilities []*ecs.Attribute, tags []*ecs.Tag, registrationToken string) error {
-	_, _, err := client.RegisterContainerInstance(agent.containerInstanceARN, capabilities, tags, registrationToken)
+	_, availabilityZone, err := client.RegisterContainerInstance(agent.containerInstanceARN, capabilities, tags, registrationToken)
+	//set az to agent
+	agent.availabilityZone = availabilityZone
```

Comment:
do we need to call stateManager save here?

Reply:
I think we don't need to save to StateManager since az not retrieved from there,  az will update each time we call RCI.